File: app/src/main/java/com/maherhanna/cheeta/ChessboardMoves.java
package com.maherhanna.cheeta;

import java.util.ArrayList;

public class ChessboardMoves {
    private final ArrayList<Move> moves;

    public ChessboardMoves(){
        moves = new ArrayList<>();
    }

    public ChessboardMoves(ChessboardMoves copy){
        this.moves = new ArrayList<Move>(copy.moves);
    }

    public void add(Move move) {
        this.moves.add(move);
    }
    public void removeLastMove(){
        moves.remove(moves.size() -1);
    }


    public Move getLastMove(){
        if(moves.size() == 0){return null;}
        return moves.get(moves.size() -1);
    }

    public Piece.Color getToPlayNow(){
        return getLastPlayed().getOpposite();
    }
    public Piece.Color getLastPlayed(){
        if(moves.size() == 0){
            return Piece.Color.BLACK;
        } else{
            return getLastMove().getColor();

        }
    }


    public boolean notEmpty() {
        return moves.size() != 0;
    }
    public int size(){return moves.size();}
    public Move get(int index){return moves.get(index);}
}

File: app/src/main/java/com/maherhanna/cheeta/ChessboardMoves.kt
package com.maherhanna.cheeta

import com.maherhanna.cheeta.Piece.Companion.GetOppositeColor

class ChessboardMoves {
    private val moves: ArrayList<Move>

    constructor() {
        moves = ArrayList()
    }

    constructor(copy: ChessboardMoves) {
        moves = ArrayList(copy.moves)
    }

    fun add(move: Move) {
        moves.add(move)
    }

    fun removeLastMove() {
        moves.removeAt(moves.size - 1)
    }

    val lastMove: Move?
        get() = if (moves.size == 0) {
            null
        } else moves[moves.size - 1]
    val toPlayNow: Int
        get() = GetOppositeColor(lastPlayed)
    val lastPlayed: Int
        get() = if (moves.size == 0) {
            Piece.BLACK
        } else {
            lastMove!!.color
        }

    fun notEmpty(): Boolean {
        return moves.size != 0
    }

    fun size(): Int {
        return moves.size
    }

    operator fun get(index: Int): Move {
        return moves[index]
    }
}
